mlflow/transformers/__init__.py

```diff
@@ -538,6 +533,14 @@ def save_model(
         else:
             mlflow_model.metadata = {FlavorKey.PROMPT_TEMPLATE: prompt_template}
 
+    if is_peft_model(built_pipeline.model):
+        _logger.info(
+            "TBA. Overriding save_pretrained to False for PEFT models,",
+            "following the Transformers behavior.",
+        )
+        save_peft_adaptor(path, built_pipeline.model)
+        save_pretrained = False
```

Comment:
I think the behavior of `save_pretrained()` for peft model is only saving the peft configs + lora weights. Basically `peft_model.save_pretrained()` will do the same job: https://huggingface.co/docs/peft/en/package_reference/peft_model#peft.PeftModel.save_pretrained

Reply:
Yea indeed `save_peft_adaptor` does the exact same job (as you pointed below). The rest of the process with `save_pretrained=False` is more for metadata logging, such as the repo and commit hash for the base model (while PEFT config stores the base model repo, we need to log commit hash in MLModel as well for security requirement).